owtf/webapp/src/containers/TargetsPage/index.js

```diff
@@ -1,19 +1,268 @@
+        targetUrls.push("http://" + line);
+        targetUrls.push("https://" + line);
+        this.setState({ newTargetUrls: "" });
+      } else {
+        this.handleAlertMsg("danger", line + " is not a valid url");
+      }
+    });
+
+    // Since we only have valid urls now in targetUrls, add them using api
+    // Proceed only if there is atleast one valid url
+    if (targetUrls.length > 0) {
+      this.handleAlertMsg("info", "Targets are being added in the background, and will appear in the table soon"); 
+      targetUrls.map(target_url => {
+        this.props.onCreateTarget(target_url);
+        setTimeout(()=> {
+          if(this.props.createError !== false){
+            this.handleAlertMsg("danger", "Unable to add " + unescape(target_url.split("//")[1]));
+          }
+        }, 200);
+      });
+    }
+    this.setState({ disabled: false });
+  }
+
+  componentDidMount() {
+    this.props.onFetchTarget();
+  }
+
+  //return current active session
+  getCurrentSession() {
+    return this.refs.sessionReference.getWrappedInstance().getCurrentSession()
+  }
+
+  //download list of targets as txt file
+  exportTargets() {
+    const targetsArray = [];
+    this.props.targets.map((target) => {
+      targetsArray.push(target.target_url+'\n');
+    });
+    const element = document.createElement("a");
+    const file = new Blob(targetsArray, {type: 'text/plain;charset=utf-8'});
+    element.href = URL.createObjectURL(file);
+    element.download = "targets.txt";
+    element.click();
   }
 
   render() {
+    const { targets, fetchLoading, fetchError } = this.props;
+    const TargetsTableProps = {
+      targets: targets,
+      getCurrentSession: this.getCurrentSession,
+      handleAlertMsg: this.handleAlertMsg,
+    }
     return (
-      <Sessions />
+      <Grid>
+        <Row>
+          <Col>{this.renderAlert()}</Col>
+        </Row>
+        <Row>
+          <Breadcrumb>
+            <Breadcrumb.Item href="/">Home</Breadcrumb.Item>
+            <Breadcrumb.Item active>Targets</Breadcrumb.Item>
+          </Breadcrumb>
+        </Row>
+        <Row>
+          <Col xs={12} sm={12} md={6} lg={6}>
+            <Row>
+              <h3>Add Targets</h3>
+            </Row>
+            <Row>
+              <FormControl
+                componentClass="textarea"
+                name="newTargetUrls"
+                placeholder="Targets seperated by new line"
+                onChange={this.handleTargetUrlsChange}
+                value={this.state.newTargetUrls}
+              />
+            </Row>
+            <Row className="add-target-btn">
+              <Button bsStyle="primary" disabled={this.state.disabled} onClick={this.addNewTargets}>
+                Add Targets
+              </Button>
+            </Row>
+          </Col>
+          <Col xs={12} sm={12} md={6} lg={6}>
+            <Row>
+              <Col xs={6} md={6}>
+                <Sessions ref="sessionReference" />
```

Comment:
@sharmamohit123 I don't think so you need to refs for this case. You can selector to getCurrentSession. That's all the reason for using store.